repair/repair.cc

```diff
@@ -1989,26 +2024,81 @@ future<> repair_service::do_rebuild_replace_with_repair(locator::token_metadata_
             }
             auto& strat = erm->get_replication_strategy();
             dht::token_range_vector ranges = strat.get_ranges(myid, *tmptr).get();
-            auto& topology = erm->get_token_metadata().get_topology();
             std::unordered_map<dht::token_range, repair_neighbors> range_sources;
             auto nr_tables = get_nr_tables(db, keyspace_name);
-            rlogger.info("{}: started with keyspace={}, source_dc={}, nr_ranges={}, ignore_nodes={}", op, keyspace_name, source_dc, ranges.size() * nr_tables, ignore_nodes);
+            sstring source_dc_for_keyspace;
+            sstring alt_dc_for_keyspace;
+            auto find_alternative_datacenter = [&] {
+                std::vector<sstring> dcs;
+                dcs.reserve(live_nodes_per_dc.size());
+                std::ranges::copy_if(topology.get_datacenters(), std::back_inserter(dcs), [&] (const auto& dc) {
+                    return dc != *source_dc && !live_nodes_per_dc[dc].empty();
```

Comment:
This is only selecting dc with live nodes. We are supposed to chose dc with zero lost nodes with higher priority.

Reply:
We can sort dc by number of lost nodes and prefer the dc with least lost nodes.

Reply:
Yes.  I'd still keep shuffling and stable sort so to distribute the load more evenly across keyspace.